### ISSUE TITLE
Call for Contributions: Add languages/translations to Material

### ISSUE THREAD

**Issue by squidfunk** [squidfunk/mkdocs-material#509]
## Call for Contributions: translating Material for MkDocs

This is a call for contributions to add translations to Material for MkDocs. Default translations are a much requested feature. #507 finally adds the ability to set the language via `mkdocs.yml`:

``` yml
theme:
  language: 'de'
```

Our progress so far:

* [x] English `en` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/en.html))
* [x] German `de` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/de.html))
* [x] Danish `da` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/da.html)) thanks to @jonas 
* [x] Polish `pl` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/pl.html)) thanks to @makshh
* [x] Swedish `sv` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/sv.html)) thanks to @martinbira 
* [x] Spanish `es` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/es.html)) thanks to @aliciaphes 
* [x] Portguese `pt` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/pt.html))  thanks to @argenta-web
* [x] French `fr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/fr.html)) thanks to @mvertes
* [x] Italian `it` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/it.html)) thanks to @markainick
* [x] Norwegian `no` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/no.html)) thanks to @oddcb
* [x] Chinese (Simplified) `zh` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/zh.html)) thanks to @cxlwill
* [x] Chinese (Traditional) `zh-Hant` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/zh-Hant.html)) thanks to @cxlwill
* [x] Korean `kr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/kr.html)) thanks to @ChoJH
* [x] Russian `ru` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/ru.html)) thanks to @leonidboykov
* [x] Japanese `ja` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/ja.html)) thanks to @miyahan
* [x] Dutch `nl` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/nl.html)) thanks to @Elvander
* [x] Turkish `tr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/tr.html)) thanks to @gurolg
* [x] Ukrainian `uk` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/uk.html)) thanks to @Serg2294 
* [ ] Finnish `fi`
* [ ] Romanian `ro`
* [ ] Hungarian `hu`
* [ ] Feel free to add more languages...

### Option 1: submit a PR
 
Feel free to add your language in a PR by copying the English translations and adjusting the labels:

``` sh
cp partials/language/en.html partials/language/ru.html 
```

Then adjust all translations to match your language:

``` jinja
{% macro t(key) %}{{ {
  "language": "ru",
  ...
}[key] }}{% endmacro %}
```

Then build the theme and submit a PR.

### Option 2: post translations in this thread

Alternatively just post the translations in this thread starting with:

``` jinja
<!-- Translations: English -->
{% macro t(key) %}{{ {
  "language": "en",
  "clipboard.copy": "Copy to clipboard",
  "clipboard.copied": "Copied to clipboard",
  "edit.link.title": "Edit this page",
  "footer.previous": "Previous",
  "footer.next": "Next",
  "meta.comments": "Comments",
  "meta.source": "Source",
  "search.placeholder": "Search",
  "search.result.placeholder": "Type to start searching",
  "search.result.none": "No matching documents",
  "search.result.one": "1 matching document",
  "search.result.other": "# matching documents",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "Go to repository",
  "toc.title": "Table of contents"
}[key] }}{% endmacro %}
```

**Comment by mvertes** [comment]
French translation below

```
<!-- Translations: English -->
{% macro t(key) %}{{ {
  "language": "fr",
  "clipboard.copy": "Copier dans le presse-papier",
  "clipboard.copied": "Copié dans le presse-papier",
  "edit.link.title": "Editer cette page",
  "footer.previous": "Précédent",
  "footer.next": "Suivant",
  "meta.comments": "Commentaires",
  "meta.source": "Source",
  "search.placeholder": "Rechercher",
  "search.result.placeholder": "Entrer pour démarrer la recherche",
  "search.result.none": "Aucun document trouvé",
  "search.result.one": "1 document trouvé",
  "search.result.other": "# documents trouvés",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "Aller au dépôt",
  "toc.title": "Table des matières"
}[key] }}{% endmacro %}
```

**Comment by ChoJH** [comment]
Below is the Korean translation.

```
<!-- Translations: Korean -->
{% macro t(key) %}{{ {
  "language": "kr",
  "clipboard.copy": "클립보드로 복사",
  "clipboard.copied": "클립보드에 복사됨",
  "edit.link.title": "이 페이지를 편집",
  "footer.previous": "이전",
  "footer.next": "다음",
  "meta.comments": "댓글",
  "meta.source": "출처",
  "search.placeholder": "검색",
  "search.result.placeholder": "검색어를 입력하세요",
  "search.result.none": "검색어와 일치하는 문서가 없습니다",
  "search.result.one": "1개의 일치하는 문서",
  "search.result.other": "#개의 일치하는 문서",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "저장소로 이동",
  "toc.title": "목차"
}[key] }}{% endmacro %}
```

**Comment by leonidboykov** [comment]
Russian translation.

```
<!-- Translation: Russian -->
{% macro t(key) %}{{ {
  "language": "ru",
  "clipboard.copy": "Копировать в буфер",
  "clipboard.copied": "Скопировано в буфер",
  "edit.link.title": "Редактировать страницу",
  "footer.previous": "Назад",
  "footer.next": "Вперед",
  "meta.comments": "Комментарии",
  "meta.source": "Исходный код",
  "search.placeholder": "Поиск",
  "search.result.placeholder": "Начните печатать для поиска",
  "search.result.none": "Совпадений не найдено",
  "search.result.one": "Найдено 1 совпадение",
  "search.result.other": "Найдено # совпадений",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "Перейти к репозиторию",
  "toc.title": "Содержание"
}[key] }}{% endmacro %}
```

**Comment by argenta-web** [comment]
Brazilian Portuguese translation:


{% macro t(key) %}{{ {
  "language": "pt",
  "clipboard.copy": "Copiar para área de transferência",
  "clipboard.copied": "Copiado para área de transferência",
  "edit.link.title": "Editar esta página",
  "footer.previous": "Anterior",
  "footer.next": "Próximo",
  "meta.comments": "Comentários",
  "meta.source": "Fonte",
  "search.placeholder": "Buscar",
  "search.result.placeholder": "Digite para iniciar a busca",
  "search.result.none": "Nenhum resultado encontrado",
  "search.result.one": "1 resultado encontrado",
  "search.result.other": "# resultados encontrados",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "Ir ao repositório",
  "toc.title": "Índice"
}[key] }}{% endmacro %}

**Comment by squidfunk** [comment]
#666 implements skip links, so we have a new phrase that needs to be translated:

English:

```
Skip to content
```

German:

```
Zum Inhalt
```

I would kindly ask all contributors to translate the term into their respective language inside this thread. I will add the translations to the theme.

* [x] English `en` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/en.html))
* [x] German `de` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/de.html))
* [ ] Danish `da` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/da.html)) thanks to @jonas 
* [ ] Polish `pl` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/pl.html)) thanks to @makshh
* [ ] Swedish `sv` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/sv.html)) thanks to @martinbira 
* [ ] Spanish `es` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/es.html)) thanks to @aliciaphes 
* [ ] Portguese `pt` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/pt.html))  thanks to @argenta-web
* [ ] French `fr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/fr.html)) thanks to @mvertes
* [ ] Italian `it` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/it.html)) thanks to @markainick
* [ ] Norwegian `no` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/no.html)) thanks to @oddcb
* [ ] Chinese (Simplified) `zh` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/zh.html)) thanks to @cxlwill
* [ ] Chinese (Traditional) `zh-Hant` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/zh-Hant.html)) thanks to @cxlwill
* [ ] Korean `kr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/kr.html)) thanks to @ChoJH
* [ ] Russian `ru` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/ru.html)) thanks to @leonidboykov
* [ ] Japanese `ja` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/ja.html)) thanks to @miyahan
* [ ] Dutch `nl` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/nl.html)) thanks to @Elvander
* [ ] Turkish `tr` ([Source](https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/language/tr.html)) thanks to @gurolg

**Comment by oddcb** [comment]
Norwegian (skip to content): Gå til innholdet

**Comment by martinbira** [comment]
Swedish (skip to content): Gå till innehållet

**Comment by jonas** [comment]
Danish: Gå til indholdet
-- 
Jonas Fonseca

**Comment by cxlwill** [comment]
Simplified Chinese: 跳转至
Traditional Chinese:  跳轉至

**Comment by leonidboykov** [comment]
Russian (skip to content): Перейти к содержанию

**Comment by argenta-web** [comment]
Portuguese: Ir para o conteúdo

--
Marco André Argenta, Dr.
Professor Adjunto - Departamento de Construção Civil
Programa de Pós-Graduação em Engenharia de Construção Civil - PPGECC
Universidade Federal do Paraná - UFPR
www.estruturas.ufpr.br
www.ppgcc.ufpr.br
www.dcc.ufpr.br
+55 41 3361-3447/3218

2018-01-18 18:57 GMT-02:00 Martin Donath <notifications@github.com>:

**Comment by markainick** [comment]
italian: ```Vai al contenuto```

**Comment by Elvander** [comment]
Dutch: `Ga naar inhoud`

**Comment by Serg2294** [comment]
Ukrainian translation below

```html
<!-- Translation: Ukrainian -->
{% macro t(key) %}{{ {
  "language": "ua",
  "clipboard.copy": "Скопіювати в буфер",
  "clipboard.copied": "Скопійовано в буфер",
  "edit.link.title": "Редагувати сторінку",
  "footer.previous": "Назад",
  "footer.next": "Вперед",
  "meta.comments": "Коментарі",
  "meta.source": "Вихідний код",
  "search.placeholder": "Пошук",
  "search.result.placeholder": "Розпочніть писати для пошуку",
  "search.result.none": "Збігів не знайдено",
  "search.result.one": "Знайдено 1 збіг",
  "search.result.other": "Знайдено # збігів",
  "search.tokenizer": "[\s\-]+",
  "source.link.title": "Перейти до репозиторію",
  "toc.title": "Зміст"
}[key] }}{% endmacro %}
```

Ukrainian (skip to content): `перейти до змісту`

**Comment by squidfunk** [comment]
All skip to content translations were already included with 2.5.1, Ukrainian is about to be released in a second with 2.5.3. Thanks guys!

**Comment by squidfunk** [comment]
I'm closing this issue and would linke to thank all contributors for a jump start in translating the theme. Some skip-to-content link translations are still missing, but that's not a big deal because Material now always falls back to English.

I added an Admonition to the landing page of the documentation which points to a [pre-filled GitHub issue](http://bit.ly/2EbzFc8). Please use this approach to submit new translations. I think this approach should work best.
